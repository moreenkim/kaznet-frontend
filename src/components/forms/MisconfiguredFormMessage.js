import React, { Component } from "react";

import { ONA_WEBSITE, WEBSITE_NAME } from "../../constants";

export default class MisconfiguredFormMessage extends Component {
  render() {
    return (
      <div>
        <p>
          This form may not appear correctly in the {WEBSITE_NAME} mobile app.
          Additionally, it <strong>may not be possible to submit data</strong>{" "}
          to this form. Please{" "}
          <a href={ONA_WEBSITE} target="_blank">
            click this link to fix this form
          </a>. You can fix it be ensuring that:
        </p>
        <ol>
          <li>The form belongs to the right organisation</li>
          <li>
            The form exists in a project which allows its members to make form
            submissions
          </li>
        </ol>
      </div>
    );
  }
}
