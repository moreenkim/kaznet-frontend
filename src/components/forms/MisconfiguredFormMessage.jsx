import React from 'react';

import { ONA_WEBSITE, WEBSITE_NAME } from '../../constants';

function MisconfiguredFormMessage() {
  return (
    <div>
      <p>
        This form may not appear correctly in the&nbsp;
        {WEBSITE_NAME}
        mobile app. Additionally, it
        <strong>&nbsp;may not be possible to submit data&nbsp;</strong>
        to this form. Please&nbsp;
        <a href={`${ONA_WEBSITE}`} target="_blank" rel="noopener noreferrer">
          click this link to fix this form
        </a>
        . You can fix it be ensuring that:
      </p>
      <ol>
        <li>The form belongs to the right organisation</li>
        <li>The form exists in a project which allows its members to make form submissions</li>
      </ol>
    </div>
  );
}

export default MisconfiguredFormMessage;
