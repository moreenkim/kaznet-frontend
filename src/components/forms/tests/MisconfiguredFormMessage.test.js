// test ExportModal
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import MisconfiguredFormMessage from "../MisconfiguredFormMessage";

describe("components/forms/MisconfiguredFormMessage", () => {
  it("renders without crashing", () => {
    shallow(<MisconfiguredFormMessage />);
  });
  it("renders the MisconfiguredFormMessage component", () => {
    const wrapper = mount(<MisconfiguredFormMessage />);

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
