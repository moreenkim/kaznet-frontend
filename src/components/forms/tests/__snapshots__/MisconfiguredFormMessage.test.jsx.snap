// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/forms/MisconfiguredFormMessage renders the MisconfiguredFormMessage component 1`] = `
<MisconfiguredFormMessage>
  <div>
    <p>
      This form may not appear correctly in the 
      Kaznet
       mobile app. Additionally, it
      <strong>
         may not be possible to submit data 
      </strong>
      to this form. Please 
      <a
        href="https://ona.io"
        rel="noopener noreferrer"
        target="_blank"
      >
        click this link to fix this form
      </a>
      . You can fix it be ensuring that:
    </p>
    <ol>
      <li>
        The form belongs to the right organisation
      </li>
      <li>
        The form exists in a project which allows its members to make form submissions
      </li>
    </ol>
  </div>
</MisconfiguredFormMessage>
`;
